In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
data = pd.read_csv('dataset.csv')
data.head()

group  (control:0, AMI:1)  \
0                          0   
1                          0   
2                          0   
3                          0   
4                          0   

   sub-type for AMI( STEMI:0, NON-STEMI :1, control:2)   Age  \
0                                                  2    28.0   
1                                                  2    40.0   
2                                                  2    64.0   
3                                                  2    57.0   
4                                                  2    58.0   

   Gender (0:female, 1:male)    WBC   RBC   HGB    BA    EO    LY  ...    MPV  \
0                          1  10.90  5.09  15.3  0.06  0.19  2.77  ...  10.00   
1                          0   5.96  5.16  15.6  0.03  0.37  2.36  ...  13.30   
2                          1   4.59  5.51  14.9  0.05  0.05  1.12  ...  11.00   
3                          0   7.26  4.70  12.5  0.05  0.11  3.46  ...  10.00   
4                          0   7.15  4.37  12.1  0.03  0.34  1.90  ...   9.69   

     PLT   HCT   MCV  RDW-SD  RDW-CV   MCH  MCHC  PDW    PCT  
0  309.0  46.3  91.9    58.0    11.5  30.1  33.0  11.2  0.31  
1  214.0  45.9  89.0    40.4    12.4  30.2  34.0  22.0  0.28  
2  193.0  43.8  79.5    38.2    13.4  27.0  34.0  13.6  0.21  
3  321.0  40.2  85.5    46.0    14.8  24.6  32.1  11.9  0.32  
4  209.0  38.0  87.0    48.0    15.1  37.7  31.8  11.4  0.20  

[5 rows x 26 columns]

ubah nama kolomnya

In [3]:
data = data.rename(columns={'sub-type for AMI( STEMI:0, NON-STEMI :1, control:2)': 'sub-type for AMI'})
data = data.rename(columns={'Gender (0:female, 1:male)':'Gender'})
data

group  (control:0, AMI:1)  sub-type for AMI   Age  Gender    WBC   RBC  \
0                            0                 2  28.0       1  10.90  5.09   
1                            0                 2  40.0       0   5.96  5.16   
2                            0                 2  64.0       1   4.59  5.51   
3                            0                 2  57.0       0   7.26  4.70   
4                            0                 2  58.0       0   7.15  4.37   
..                         ...               ...   ...     ...    ...   ...   
976                          1                 1  43.0       1  10.09  5.25   
977                          1                 1  58.0       1  14.32  5.34   
978                          1                 0  62.0       1   9.64  4.26   
979                          1                 1  48.0       0  11.49  4.05   
980                          1                 1  58.0       0   7.30  4.07   

      HGB    BA    EO    LY  ...    MPV    PLT    HCT   MCV  RDW-SD  RDW-CV  \
0    15.3  0.06  0.19  2.77  ...  10.00  309.0  46.30  91.9    58.0    11.5   
1    15.6  0.03  0.37  2.36  ...  13.30  214.0  45.90  89.0    40.4    12.4   
2    14.9  0.05  0.05  1.12  ...  11.00  193.0  43.80  79.5    38.2    13.4   
3    12.5  0.05  0.11  3.46  ...  10.00  321.0  40.20  85.5    46.0    14.8   
4    12.1  0.03  0.34  1.90  ...   9.69  209.0  38.00  87.0    48.0    15.1   
..    ...   ...   ...   ...  ...    ...    ...    ...   ...     ...     ...   
976  15.6  0.03  0.11  3.03  ...   8.50  259.0  43.20  82.3    41.0    13.8   
977  15.0  0.11  0.31  3.33  ...  10.10  227.0  43.60  86.0    38.2    13.1   
978  13.9  0.05  0.01  0.83  ...  10.90  171.0  39.50  92.5    43.5    12.8   
979  12.6  0.12  0.06  3.55  ...   9.60  309.0  37.79  93.1    40.4    12.0   
980  12.1  0.03  0.07  2.25  ...   9.40  182.0  34.40  84.5    41.1    13.3   

      MCH   MCHC  PDW    PCT  
0    30.1  33.00  11.2  0.31  
1    30.2  34.00  22.0  0.28  
2    27.0  34.00  13.6  0.21  
3    24.6  32.10  11.9  0.32  
4    37.7  31.80  11.4  0.20  
..    ...    ...   ...   ...  
976  29.7  36.10  15.8  0.22  
977  28.1  34.40  11.6  0.23  
978  32.6  35.20  13.4  0.19  
979  31.0  33.29  10.1  0.30  
980  29.7  35.20  10.3  0.17  

[981 rows x 26 columns]

In [4]:
data = data[['sub-type for AMI', 'Age', 'Gender','WBC',
       'RBC', 'HGB', 'BA', 'EO', 'LY', 'MO', 'NEU', 'NEU/LY', 'PLT/LY',
       'MPV/LY', 'LY/MO', 'MPV', 'PLT', 'HCT', 'MCV', 'RDW-SD', 'RDW-CV',
       'MCH', 'MCHC', 'PDW ', 'PCT']].copy()
data

sub-type for AMI   Age  Gender    WBC   RBC   HGB    BA    EO    LY  \
0                   2  28.0       1  10.90  5.09  15.3  0.06  0.19  2.77   
1                   2  40.0       0   5.96  5.16  15.6  0.03  0.37  2.36   
2                   2  64.0       1   4.59  5.51  14.9  0.05  0.05  1.12   
3                   2  57.0       0   7.26  4.70  12.5  0.05  0.11  3.46   
4                   2  58.0       0   7.15  4.37  12.1  0.03  0.34  1.90   
..                ...   ...     ...    ...   ...   ...   ...   ...   ...   
976                 1  43.0       1  10.09  5.25  15.6  0.03  0.11  3.03   
977                 1  58.0       1  14.32  5.34  15.0  0.11  0.31  3.33   
978                 0  62.0       1   9.64  4.26  13.9  0.05  0.01  0.83   
979                 1  48.0       0  11.49  4.05  12.6  0.12  0.06  3.55   
980                 1  58.0       0   7.30  4.07  12.1  0.03  0.07  2.25   

       MO  ...    MPV    PLT    HCT   MCV  RDW-SD  RDW-CV   MCH   MCHC  PDW   \
0    0.96  ...  10.00  309.0  46.30  91.9    58.0    11.5  30.1  33.00  11.2   
1    0.43  ...  13.30  214.0  45.90  89.0    40.4    12.4  30.2  34.00  22.0   
2    0.27  ...  11.00  193.0  43.80  79.5    38.2    13.4  27.0  34.00  13.6   
3    0.70  ...  10.00  321.0  40.20  85.5    46.0    14.8  24.6  32.10  11.9   
4    0.47  ...   9.69  209.0  38.00  87.0    48.0    15.1  37.7  31.80  11.4   
..    ...  ...    ...    ...    ...   ...     ...     ...   ...    ...   ...   
976  0.52  ...   8.50  259.0  43.20  82.3    41.0    13.8  29.7  36.10  15.8   
977  1.45  ...  10.10  227.0  43.60  86.0    38.2    13.1  28.1  34.40  11.6   
978  0.39  ...  10.90  171.0  39.50  92.5    43.5    12.8  32.6  35.20  13.4   
979  0.63  ...   9.60  309.0  37.79  93.1    40.4    12.0  31.0  33.29  10.1   
980  0.57  ...   9.40  182.0  34.40  84.5    41.1    13.3  29.7  35.20  10.3   

      PCT  
0    0.31  
1    0.28  
2    0.21  
3    0.32  
4    0.20  
..    ...  
976  0.22  
977  0.23  
978  0.19  
979  0.30  
980  0.17  

[981 rows x 25 columns]

cek ada tidaknya NULL value

In [5]:
print(data.isnull().sum())

sub-type for AMI    0
Age                 0
Gender              0
WBC                 0
RBC                 0
HGB                 0
BA                  0
EO                  0
LY                  0
MO                  0
NEU                 0
NEU/LY              0
PLT/LY              0
MPV/LY              0
LY/MO               0
MPV                 0
PLT                 0
HCT                 0
MCV                 0
RDW-SD              0
RDW-CV              0
MCH                 0
MCHC                0
PDW                 0
PCT                 0
dtype: int64


In [6]:
#statistik deskriptif dari dataset
data.describe()

sub-type for AMI         Age      Gender         WBC         RBC  \
count        981.000000  981.000000  981.000000  981.000000  981.000000   
mean           1.328236   59.166157    0.579001    9.091019    4.755474   
std            0.769513   10.851216    0.493971    3.201733    0.546632   
min            0.000000   20.000000    0.000000    3.000000    2.700000   
25%            1.000000   51.000000    0.000000    6.980000    4.380000   
50%            2.000000   60.000000    1.000000    8.500000    4.750000   
75%            2.000000   67.000000    1.000000   10.520000    5.140000   
max            2.000000   82.000000    1.000000   30.900000    7.220000   

              HGB          BA          EO          LY          MO  ...  \
count  981.000000  981.000000  981.000000  981.000000  981.000000  ...   
mean    13.747441    0.056177    0.163639    2.490805    0.659949  ...   
std      1.576756    0.062927    0.141448    1.175224    0.260870  ...   
min     10.080000    0.010000    0.010000    0.400000    0.040000  ...   
25%     12.600000    0.020000    0.080000    1.800000    0.490000  ...   
50%     13.700000    0.050000    0.120000    2.290000    0.600000  ...   
75%     14.900000    0.080000    0.210000    2.920000    0.780000  ...   
max     17.890000    1.100000    1.150000    9.000000    1.900000  ...   

              MPV         PLT         HCT         MCV      RDW-SD      RDW-CV  \
count  981.000000  981.000000  981.000000  981.000000  981.000000  981.000000   
mean     9.817655  259.752294   41.048981   86.699021   41.864730   13.670275   
std      3.450854   71.165335    4.357617    5.635603    4.041242    1.370960   
min      6.100000  100.000000   19.400000   61.300000   18.100000    9.600000   
25%      8.900000  210.000000   38.000000   83.900000   39.000000   12.800000   
50%      9.800000  256.000000   41.100000   86.800000   41.000000   13.400000   
75%     10.500000  298.000000   44.200000   89.800000   43.800000   14.200000   
max    110.400000  754.000000   54.000000  120.000000   68.400000   22.300000   

              MCH        MCHC        PDW          PCT  
count  981.000000  981.000000  981.000000  981.000000  
mean    29.004944   33.433721   13.499235    0.254485  
std      2.390958    1.545107    2.927959    0.135490  
min     18.100000   23.600000    7.600000    0.050000  
25%     27.900000   32.500000   11.100000    0.200000  
50%     29.200000   33.500000   12.800000    0.240000  
75%     30.400000   34.400000   16.390000    0.280000  
max     44.100000   55.290000   23.100000    2.890000  

[8 rows x 25 columns]

In [7]:
data.dtypes

sub-type for AMI      int64
Age                 float64
Gender                int64
WBC                 float64
RBC                 float64
HGB                 float64
BA                  float64
EO                  float64
LY                  float64
MO                  float64
NEU                 float64
NEU/LY              float64
PLT/LY              float64
MPV/LY              float64
LY/MO               float64
MPV                 float64
PLT                 float64
HCT                 float64
MCV                 float64
RDW-SD              float64
RDW-CV              float64
MCH                 float64
MCHC                float64
PDW                 float64
PCT                 float64
dtype: object

In [8]:
data.shape

(981, 25)

In [9]:
print(type(data))

<class 'pandas.core.frame.DataFrame'>


melakukan pengujian pada distribusi data menggunakan shapiro-wilk

In [10]:
from scipy import stats
#fungsi untuk melakukan uji shapiro-wilk pada semua kolom
def shapiro_wilk_test(data):
    results = {}
    for column in data.columns:
        stat, p_value = stats.shapiro(data[column].dropna())  # dropna() untuk menghindari null values
        results[column] = {'statistic': stat, 'p_value': p_value}
    return results

#menjalankan uji dan menampilkan hasilnya
shapiro_results = shapiro_wilk_test(data)
#menampilkan hasil uji untuk setiap kolom
for column, result in shapiro_results.items():
    print(f"Uji Shapiro-Wilk untuk {column}: Statistik = {result['statistic']}, P-value = {result['p_value']}")
    if result['p_value'] > 0.05:
        print(f"{column}: Data terdistribusi normal (gagal menolak hipotesis nol)\n")
    else:
        print(f"{column}: Data tidak terdistribusi normal (menolak hipotesis nol)\n")

Uji Shapiro-Wilk untuk sub-type for AMI: Statistik = 0.7499774098396301, P-value = 4.851721995690485e-36
sub-type for AMI: Data tidak terdistribusi normal (menolak hipotesis nol)

Uji Shapiro-Wilk untuk Age: Statistik = 0.9893985986709595, P-value = 1.5934265320538543e-06
Age: Data tidak terdistribusi normal (menolak hipotesis nol)

Uji Shapiro-Wilk untuk Gender: Statistik = 0.6274404525756836, P-value = 1.2384675827702733e-41
Gender: Data tidak terdistribusi normal (menolak hipotesis nol)

Uji Shapiro-Wilk untuk WBC: Statistik = 0.9069954752922058, P-value = 5.5064869600055865e-24
WBC: Data tidak terdistribusi normal (menolak hipotesis nol)

Uji Shapiro-Wilk untuk RBC: Statistik = 0.996276319026947, P-value = 0.019254593178629875
RBC: Data tidak terdistribusi normal (menolak hipotesis nol)

Uji Shapiro-Wilk untuk HGB: Statistik = 0.9918331503868103, P-value = 3.0001589038874954e-05
HGB: Data tidak terdistribusi normal (menolak hipotesis nol)

Uji Shapiro-Wilk untuk BA: Statistik = 0.4

feature selection: filter method

In [11]:
from scipy.stats import spearmanr

# Memisahkan fitur X dan target y
X = data.drop(columns=['sub-type for AMI'])
y = data['sub-type for AMI']

# Menghitung korelasi Spearman
spearman = {}
for kolom in X.columns:
    # Mengambil korelasi Spearman antara fitur dan target
    correlation, _ = spearmanr(X[kolom], y)  # Hanya mengambil korelasi, bukan p-value
    spearman[kolom] = correlation

# Menyortir fitur berdasarkan nilai korelasi dengan abs()
fitur_tersortir = sorted(spearman.items(), key=lambda x: abs(x[1]), reverse=True)

# Menampilkan fitur dan nilai korelasi
for feature, corr in fitur_tersortir:
    print(f"{feature}: {corr}")


WBC: -0.6039680651943417
NEU: -0.558550490224538
MO: -0.326899290598384
PDW : -0.3235464902809049
MPV: 0.2695681366989625
NEU/LY: -0.26157990493218747
MCHC: -0.233208499501585
RDW-CV: -0.20548534778728095
BA: -0.2037543637948184
MPV/LY: 0.18771740222898722
Gender: -0.1735842917562152
PLT/LY: 0.12912836957815513
LY/MO: 0.12717903739758057
LY: -0.11119197234908462
PCT: 0.10577645519735131
MCH: -0.08926470044455803
HGB: -0.08823815675305229
EO: 0.08180636498642022
Age: -0.07057554882350114
MCV: 0.05227060170491408
RDW-SD: -0.04210842203099182
RBC: -0.04127932818704308
PLT: -0.02483347426181484
HCT: -0.004246851208031298


In [12]:
#menghapus fitur dengan korelasi rendah
fitur_terpilih = {k: v for k, v in spearman.items() if abs(v) >= 0.3}
print("fitur terpilih: ",fitur_terpilih)

fitur terpilih:  {'WBC': -0.6039680651943417, 'MO': -0.326899290598384, 'NEU': -0.558550490224538, 'PDW ': -0.3235464902809049}


In [13]:
data = data[['sub-type for AMI','WBC','MO','NEU','PDW ']].copy()
data

sub-type for AMI    WBC    MO   NEU  PDW 
0                   2  10.90  0.96  6.22  11.2
1                   2   5.96  0.43  2.77  22.0
2                   2   4.59  0.27  3.10  13.6
3                   2   7.26  0.70  2.94  11.9
4                   2   7.15  0.47  4.41  11.4
..                ...    ...   ...   ...   ...
976                 1  10.09  0.52  6.40  15.8
977                 1  14.32  1.45  9.11  11.6
978                 0   9.64  0.39  8.36  13.4
979                 1  11.49  0.63  7.30  10.1
980                 1   7.30  0.57  4.37  10.3

[981 rows x 5 columns]

mengecek apakah label balance atau tidak

In [14]:
#menghitung jumlah sample per kelas
print(data['sub-type for AMI'].value_counts())

sub-type for AMI
2    504
1    295
0    182
Name: count, dtype: int64


kelas 2 = 51.3%

kelas 1 = 30.1%

kelas 0 = 18.5%

sehingga dapat dikatakan data tidak seimbang atau imbalance

In [15]:
X = data.iloc[:,:5]
X.head()

sub-type for AMI    WBC    MO   NEU  PDW 
0                 2  10.90  0.96  6.22  11.2
1                 2   5.96  0.43  2.77  22.0
2                 2   4.59  0.27  3.10  13.6
3                 2   7.26  0.70  2.94  11.9
4                 2   7.15  0.47  4.41  11.4

In [16]:
Y = data.loc[:,['sub-type for AMI']]
Y.head()

sub-type for AMI
0                 2
1                 2
2                 2
3                 2
4                 2

membagi dataset menjadi traning set dan test set

In [17]:
from imblearn.over_sampling import SMOTE
from sklearn.model_selection import train_test_split
from sklearn import datasets

#pisahkan data menjadi traning dan testing
X_train,X_test,y_train,y_test = train_test_split(X,Y,test_size=0.3,random_state=42,stratify=Y)

balancing = SMOTE(random_state=42)
X_train_smote,y_train_smote = balancing.fit_resample(X_train,y_train)


In [18]:
len(y_train)

686

# Implementasi model 1: Logistic regression


In [19]:
# multiclass logistic regression 
from sklearn.linear_model import LogisticRegression

#membentuk model regresi logistik untuk multiclass
model = LogisticRegression(multi_class='multinomial',solver='lbfgs')

model.fit(X_train,y_train)

print(model.coef_)

[[-6.14818638e+00  2.79104417e-01  2.15754080e-01 -3.54464974e-02
   1.46708969e-01]
 [-4.27143028e-01 -3.08033524e-02 -8.15729087e-02  9.17508839e-02
   1.74767396e-03]
 [ 6.57532940e+00 -2.48301065e-01 -1.34181171e-01 -5.63043865e-02
  -1.48456643e-01]]


c:\Users\Acer\anaconda3\Lib\site-packages\sklearn\utils\validation.py:1143: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
c:\Users\Acer\anaconda3\Lib\site-packages\sklearn\linear_model\_logistic.py:458: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


In [21]:
#melakukan prediksi terhadap label
predict = model.predict(X_test)
print(predict)

#untuk  mendapatkan probabilitas prediksi untuk setiap kelas
probabilitas = model.predict_proba(X_test)
print(probabilitas)

[2 2 1 1 0 1 2 2 1 1 2 1 0 1 0 0 2 2 1 2 1 0 0 2 0 1 0 1 2 2 2 2 2 1 2 1 1
 0 2 2 0 2 2 0 0 1 1 1 1 2 2 2 0 1 1 1 2 2 2 1 1 0 2 1 1 0 1 0 2 1 2 1 2 2
 2 2 0 2 2 1 0 2 2 2 2 2 2 2 0 1 0 2 1 1 0 1 1 2 1 2 0 0 2 1 2 2 2 1 1 2 0
 0 2 2 0 2 1 2 1 2 2 0 2 2 1 0 2 2 2 1 2 1 1 2 2 1 2 2 2 1 2 2 1 0 2 2 1 1
 1 0 1 1 1 2 1 2 1 2 2 1 1 2 2 2 0 1 2 2 2 2 2 1 0 1 2 2 1 1 2 0 2 2 2 1 2
 1 2 0 0 0 2 1 1 1 2 2 2 2 2 1 2 1 2 1 1 2 2 2 0 2 1 0 2 2 2 2 0 2 2 1 2 2
 2 2 2 2 1 2 0 2 2 2 0 0 1 2 2 2 1 1 2 2 2 2 0 0 0 0 2 1 2 2 2 0 2 1 2 2 2
 2 0 1 2 2 0 2 1 2 1 1 2 2 0 1 0 2 0 1 1 1 0 0 0 2 2 2 2 2 2 2 2 2 1 0 2]
[[2.80499743e-06 3.12432303e-02 9.68753965e-01]
 [6.10911638e-07 1.32965048e-02 9.86702884e-01]
 [3.66839826e-02 9.42702324e-01 2.06136936e-02]
 [6.07616111e-02 9.30058963e-01 9.17942548e-03]
 [9.71298010e-01 2.87018187e-02 1.71477431e-07]
 [2.19788554e-02 9.51802669e-01 2.62184759e-02]
 [7.35659960e-07 1.44680082e-02 9.85531256e-01]
 [3.50434486e-07 1.35601179e-02 9.86439532e-01]
 [2.70873121e-02 

In [ ]:
from sklearn.model_selection import cross_val_score

skor = cross_val_score(model, X,Y, cv=5,scoring='accuracy')
print("cross-validation scores: ",skor)
print("mean skor: ",skor.mean())

c:\Users\Acer\anaconda3\Lib\site-packages\sklearn\utils\validation.py:1143: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
c:\Users\Acer\anaconda3\Lib\site-packages\sklearn\linear_model\_logistic.py:458: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
c:\Users\Acer\anaconda3\Lib\site-packages\sklearn\utils\validation.py:1143: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using

cross-validation scores:  [1. 1. 1. 1. 1.]
mean skor:  1.0


c:\Users\Acer\anaconda3\Lib\site-packages\sklearn\linear_model\_logistic.py:458: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
c:\Users\Acer\anaconda3\Lib\site-packages\sklearn\utils\validation.py:1143: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
c:\Users\Acer\anaconda3\Lib\site-packages\sklearn\linear_model\_logistic.py:458: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max

In [ ]:
#evaluasi model
from sklearn.metrics import classification_report

report = classification_report(y_test,predict)
print(report)

              precision    recall  f1-score   support

           0       0.98      1.00      0.99        55
           1       1.00      0.99      0.99        89
           2       1.00      1.00      1.00       151

    accuracy                           1.00       295
   macro avg       0.99      1.00      1.00       295
weighted avg       1.00      1.00      1.00       295



Implementasi model:

In [ ]:
from sklearn.ensemble import RandomForestClassifier

model = RandomForestClassifier(n_estimators=100,random_state=42)
model.fit(X_train,y_train)

y_pred = model.predict(X_test)

C:\Users\Acer\AppData\Local\Temp\ipykernel_10992\173348647.py:4: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  model.fit(X_train,y_train)


In [26]:
#evaluasi model
from sklearn.metrics import classification_report
print("clasification report:\n",classification_report(y_test,y_pred))

clasification report:
               precision    recall  f1-score   support

           0       1.00      1.00      1.00        55
           1       1.00      1.00      1.00        89
           2       1.00      1.00      1.00       151

    accuracy                           1.00       295
   macro avg       1.00      1.00      1.00       295
weighted avg       1.00      1.00      1.00       295

